### PR TITLE
Enable mypy type checks for the Python CDK

### PR DESCRIPTION
**Pull request by girarda**
## What
* Run mypy on modified Python CDK files
  * This was removed from the gradle build as part of the [gradle cleanup ](https://github.com/airbytehq/airbyte/pull/30060/files#diff-25d60c0116afb59ec6c764847ffb9793fc4f20a02d17cad0848ac8a13dcf5149L26-L29)
  * We're intentionally not adding this step to gradle to avoid introducing a dependency on git as it would prevent us from moving the build step to dagger
  * CDK developers can still run mypy locally using `bin/run-mypy-on-modified-files.sh`

### CODE REVIEW

**Comment by postamar** [comment]
@girarda can we instead configure `mypyCheck` to run using a whitelist or a blacklist? To be clear: I have no issue with using `run-mypy-on-modified-files.sh` for local development and/or having a gradle task wrap it. I strongly believe that you and your team should modify these tasks to suit your own needs! 

What I find problematic is that this script assumes `git` is present and available, but that's not always the case. In local development it obviously is, but `mypyCheck` is a dependency of `check` which may (soon) be run in a dagger container.

File: .github/workflows/run-mypy-on-modified-cdk-files.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Run mymy on modified cdk files
```
Comment (postamar) [inline .github/workflows/run-mypy-on-modified-cdk-files.yml:1]: s/mymy/mypy :)

Also, there might be a naming convention these days for gha, check with @airbytehq/connector-operations

File: .github/workflows/run-mypy-on-modified-cdk-files.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+on:
+  pull_request:
+    paths:
+      - "airbyte-cdk/python/airbyte_cdk/**"
```
Comment (postamar) [inline .github/workflows/run-mypy-on-modified-cdk-files.yml:6]: Does the globber support `**/*.py` ?

File: .github/workflows/run-mypy-on-modified-cdk-files.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          fetch-depth: 0
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - run: pip install mypy==1.6.0
+      - name: Get Python changed files
+        id: changed-py-files
+        uses: tj-actions/changed-files@v23
```
Comment (postamar) [inline .github/workflows/run-mypy-on-modified-cdk-files.yml:23]: neat!

**Review by postamar — APPROVED** [review]

File: .github/workflows/run-mypy-on-modified-cdk-files.yml (hunk 2 above)
Comment (girarda) [reply .github/workflows/run-mypy-on-modified-cdk-files.yml]: yes, updated!

**Pull request merged**